packages/@react-spectrum/list/src/ListView.tsx

```diff
@@ -258,7 +267,8 @@ function ListView<T extends object>(props: SpectrumListViewProps<T>, ref: DOMRef
                       <InsertionIndicator
                         key={`${item.key}-after`}
                         target={{key: item.key, type: 'item', dropPosition: 'after'}}
-                        isPresentationOnly={collection.getKeyAfter(item.key) != null} />
+                        isPresentationOnly={item.key !== lastKey}
+                        isLastRow={lastKey === item.key} />
```

Comment:
One problem is I'm not sure if this will update when a new row is added at the end, whereas it would have before via context.